Replace polygon with shapely polygon

LeonidBeynenson opened the pull request:
Note that the change was not debugged properly.

LeonidBeynenson committed:
Replace obsolete Polygon3 with shapely.geometry.Polygon

Note that the change was not debugged properly.

LeonidBeynenson committed:
Fix small bug

druzhkov-paul commented on mmdet/core/evaluation/text_evaluation.py:
```diff
@@ -16,19 +16,28 @@
-import Polygon as plg
+
+# ATTENTION! the obsolete Polygon3 was replaced with shapely.geometry.Polygon,
+#            but the changes were not debugged properly.
```
"Debugged" kinda hints that there are bugs and we know about them but won't fix. Was the logic involved tested somehow?

LeonidBeynenson replied:
Since Ilya checked the code, I removed this comment

druzhkov-paul commented on constraints.txt:
```diff
@@ -69,7 +69,6 @@ parasail==1.2.4
 Pillow==8.4.0
 pkg_resources==0.0.0
 pluggy==1.0.0
-Polygon3==3.0.8
```
Any chance to add `Shapely` to constraints?

LeonidBeynenson replied:
It is already here 🙂

Ilya-Krylov commented on mmdet/core/evaluation/text_evaluation.py:
```diff
@@ -16,19 +16,28 @@
 import numpy as np
-import Polygon as plg
+
+# ATTENTION! the obsolete Polygon3 was replaced with shapely.geometry.Polygon,
```
tested

Ilya-Krylov replied:
please remove

LeonidBeynenson replied:
Removed

LeonidBeynenson committed:
Remove unrequired comment

The comment is not required since we checked that updated code works
with shapely.geometry.Polygon properly.

druzhkov-paul commented:
Build 618 🕙
